src/index.d.ts

```diff
@@ -1,9 +1,16 @@
-export interface FluxStandardAction<Payload, Meta = undefined> {
+/**
+ * A Flux Standard action with optional payload and metadata properties.
+ */
+export interface FluxStandardAction<
+  Payload = undefined,
+  Meta = undefined,
+  Type extends string = string
```

Comment:
In TypeScript you can restrict a type to a constant value.

For a Counter application example this makes it possible to change the following code:
``` ts
import { FSA } from 'flux-standard-action';

const INCREMENT_COUNTER = 'INCREMENT_COUNTER';
type INCREMENT_COUNTER = typeof INCREMENT_COUNTER;
interface IncrementCounterAction extends FSA {
  type: INCREMENT_COUNTER;
}
```

to this:
``` ts
import { FSA } from 'flux-standard-action';

const INCREMENT_COUNTER = 'INCREMENT_COUNTER';
type IncrementCounterAction = FSA<undefined, undefined, typeof INCREMENT_COUNTER>;
```

removing the need to re-declare the `type` property. An object using a string, but different to 'INCREMENT_COUNTER' would thus not implement the `IncrementCounterAction` type.

Reply:
Why would we want to put the `Type` generic to be last? This would mean we would always have to explicitly type meta whether the action has meta or not.
I think I would prefer to put `tType` first as it is the primary property of an FSA and I want to _always_ strongly type my action types for discriminitive unions in redux reducers. Where as quite a lot of my action don't utilise the `Meta` property.

Reply:
My opinionated take on this is on the master branch of this fork if you want to have a look https://github.com/hally9k/flux-standard-action

Reply:
see https://github.com/redux-utilities/flux-standard-action/pull/114#issuecomment-482722177